### PR TITLE
AzureRT - Fix Problem of Adding Extension while using New/Set-AzureDeployment

### PR DESCRIPTION
**Pull request by huangpf**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@huangpf__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (phuang). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by huangpf** [comment]
Follow-up for PR/Fix: https://github.com/Azure/azure-powershell/pull/361

**Comment by huangpf** [comment]
Follow-up for Issue: https://github.com/Azure/azure-powershell/issues/193

File: src/ServiceManagement/Compute/Commands.ServiceManagement.Test/FunctionalTests/IaasCmdletInfo/SetAzureDeploymentCmdletInfo.cs [hunk 1]
```diff
@@ -38,12 +39,13 @@ public static SetAzureDeploymentCmdletInfo SetAzureDeploymentStatusCmdletInfo(st
...
 
 
-        public static SetAzureDeploymentCmdletInfo SetAzureDeploymentConfigCmdletInfo(string serviceName, string slot, string configPath)
+        public static SetAzureDeploymentCmdletInfo SetAzureDeploymentConfigCmdletInfo(string serviceName, string slot, string configPath, ExtensionConfigurationInput extConfig = null)
         {
             SetAzureDeploymentCmdletInfo result = new SetAzureDeploymentCmdletInfo(serviceName, slot);
 
             result.cmdletParams.Add(new CmdletParam("Config"));
             result.cmdletParams.Add(new CmdletParam("Configuration", configPath));
+            result.cmdletParams.Add(new CmdletParam("ExtensionConfiguration", extConfig));
```
Comment (hyonholee) [inline src/ServiceManagement/Compute/Commands.ServiceManagement.Test/FunctionalTests/IaasCmdletInfo/SetAzureDeploymentCmdletInfo.cs]: Are Set-AzureDeployment with "-ExtensionConfiguration $null" parameter and Set-AzureDeployment without -ExtensionConfiguration paramter the same?
If they are the same, this line should be added when 'extConfig' is not null.
If they are not the same, we need two separate methods, with $null extConfig and without extConfig.

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Extensions/Common/BaseAzureServiceExtensionCmdlet.cs [hunk 2]
```diff
@@ -285,6 +288,15 @@ protected DeploymentGetResponse GetDeployment(string slot)
             return d;
         }
 
+        protected DeploymentGetResponse GetPeerDeployment(string currentSlot)
+        {
+            var currentSlotType = (DeploymentSlot)Enum.Parse(typeof(DeploymentSlot), currentSlot, true);
+            var peerSlot = currentSlotType == DeploymentSlot.Production ? DeploymentSlot.Staging : DeploymentSlot.Production;
+            var peerSlotStr = peerSlot.ToString();
```
Comment (hyonholee) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Extensions/Common/BaseAzureServiceExtensionCmdlet.cs]: NIT: peerSlotStr is redundant.   Just call GetDeployment with peerSlot.ToString().

File: src/ServiceManagement/Compute/Commands.ServiceManagement.Test/FunctionalTests/IaasCmdletInfo/SetAzureDeploymentCmdletInfo.cs (hunk 1 above)
Comment (huangpf) [inline src/ServiceManagement/Compute/Commands.ServiceManagement.Test/FunctionalTests/IaasCmdletInfo/SetAzureDeploymentCmdletInfo.cs]: OK.

**Comment by ogail** [comment]
@huangpf please fix merge conflicts

**Comment by huangpf** [comment]
@ogail OK. Updated & waiting for the build/test.

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Extensions/Common/BaseAzureServiceExtensionCmdlet.cs (hunk 2 above)
Comment (huangpf) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Extensions/Common/BaseAzureServiceExtensionCmdlet.cs]: OK. Thanks. Will try to fix in the next PR. Now just trying to stabilize it for tests.

**Comment by huangpf** [comment]
@hyonholee Can you help sign off if no other concerns? Thanks.

**Comment by hyonholee** [comment]
LGTM

**Pull request merged**